lib/events/filelog.go

```diff
@@ -273,11 +280,12 @@ func (l *FileLog) SearchEvents(fromUTC, toUTC time.Time, namespace string, event
 		}
 
 		events = append(events, event)
+		totalSize += size
 
 		// Check if there is a limit and if so, check if we've hit it.
 		// In the event that we've hit the limit, we consider the query partially complete
 		// and return a checkpoint to continue it.
-		if len(events) >= limit && limit > 0 {
+		if (len(events) >= limit && limit > 0) || totalSize >= MaxEventBytesInResponse {
```

Comment:
nit: MaxEventBytesInResponse says that "This is the max size of all the events we return" but technically currently the limit is exceeded because size  check is done after new event is added to the `events` array.

Reply:
resolved